validator/project_validator.go

```diff
@@ -987,14 +988,18 @@ func validateProjectLimits(_ context.Context, settings *evergreen.Settings, proj
 
 // validateTaskNames ensures the task names do not contain unauthorized characters.
 func validateTaskNames(project *model.Project) ValidationErrors {
-	unauthorizedTaskCharacters := unauthorizedCharacters + " "
 	errs := ValidationErrors{}
+
+	validSpecialCharacters := []string{"_", ".", "/", ":", "•", "=", "[", "]", "-"}
```

Comment:
Should we update unauthorizedCharacters for this as well so that this impacts other places?